pkg/kubelet/kubelet.go

```diff
@@ -1615,6 +1625,30 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	}
 }
 
+// syncStatus syncs pods statuses with the apiserver.
+func (kl *Kubelet) syncStatus() {
+	glog.V(3).Infof("Syncing pods status")
+
+	for _, pod := range kl.pods {
```

Comment:
We should protect the access to kl.pods with pocLock. We should probably build a list of statuses and then send them so we don't block on network.

This also made me realize there is one case where we're not locking kl.pods, will send a PR for that. It should not affect this.